File: 1. 단계별로_풀어보기/20~29/25. 이분 탐색/2. 10816.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const cardArr = inputArr[1].trim().split(' ').map(Number)
const queryArr = inputArr[3].trim().split(' ').map(Number)

// 해시맵
const cardMap = {}

// 해시맵 저장
cardArr.forEach(card => {
  if (cardMap[card] > 0) {
    cardMap[card]++
  } // ::
  else {
    cardMap[card] = 1
  }
})

// 쿼리 수행
let resultStr = ''
queryArr.forEach(query => {
  resultStr += (cardMap[query] > 0 ? cardMap[query] : 0) + ' '
})

// 출력
process.stdout.write(resultStr)

File: 1. 단계별로_풀어보기/20~29/25. 이분 탐색/2. 10816.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())
cardArr = list(map(int, input().split()))
Q = int(input())
queryArr = list(map(int, input().split()))

cardMap = {}

for card in cardArr:
  if card in cardMap:
    cardMap[card] += 1
  else:
    cardMap[card] = 1

for query in queryArr:
  output(f"{cardMap[query] if query in cardMap else 0} ")